### PR TITLE
Fix incorrect lb endpoints for DNS ServiceEntry with workload selector

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Please provide a description for what this PR is for.

part1 fix: #26098

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: pilot/pkg/serviceregistry/serviceentry/conversion.go [hunk 1]
```diff
@@ -298,8 +298,9 @@ func convertInstances(cfg model.Config, services []*model.Service) []*model.Serv
 	}
 	for _, service := range services {
 		for _, serviceEntryPort := range serviceEntry.Ports {
-			if len(serviceEntry.Endpoints) == 0 &&
+			if len(serviceEntry.Endpoints) == 0 && serviceEntry.WorkloadSelector == nil &&
 				serviceEntry.Resolution == networking.ServiceEntry_DNS {
+				// Note: only convert the hostname to service instance if WorkloadSelector unset
```
Comment (ramaraochavali) [inline pilot/pkg/serviceregistry/serviceentry/conversion.go:303]: /s/if WorkloadSelector unset/if WorkloadSelector is not set

Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/serviceentry/conversion.go:303]: ok

**Comment by hzxuzhonghu** [comment]
@ramaraochavali Would you mind approving? I find there is a dependency on this pr when i am addressing your comments for #26112

**Comment by ramaraochavali** [comment]
sorry - I thought it was approved already

**Pull request merged**